connector/src/main/scala/com/microsoft/kusto/spark/datasink/KustoWriter.scala

```diff
@@ -189,94 +158,164 @@ object KustoWriter{
-          case OperationStatus.Succeeded =>
-            KDSU.logInfo(myName, s"Ingestion to Kusto succeeded. Cluster: '$cluster', database: '$database', table: '$table'$batchIdIfExists, partition: '$partitionId'")
-          case otherStatus =>
-            throw new RuntimeException(s"Ingestion to Kusto failed with status '$otherStatus'." +
-              s" Cluster: '$cluster', database: '$database', table: '$table'$batchIdIfExists, partition: '$partitionId'. Ingestion info: '${readIngestionResult(res)}'")
-        }).await(timeoutMillis, TimeUnit.MILLISECONDS)
-    }
+        }
+      },
+      timeout)
+  }
 
-  def All(list: util.ArrayList[Boolean]) : Boolean = {
+  def all(list: util.ArrayList[Boolean]): Boolean = {
     val it = list.iterator
     var res = true
-    while (it.hasNext && res)
-    {
+    while (it.hasNext && res) {
       res = it.next()
     }
     res
   }
 
-  @throws[IOException]
-  private[kusto] def serializeRows(rows: Iterator[InternalRow], schema: StructType, bos: BlobOutputStream, timeZone: String): Unit = {
-    val gzip = new GZIPOutputStream(bos)
+  def createFileWriter(schema: StructType,
+                       timeZone: String,
+                       tableCoordinates: KustoCoordinates,
+                       tmpTableName: String,
+                       storageUri: String,
+                       container: CloudBlobContainer): FileWriteResource = {
+    val blobName = s"${tableCoordinates.database}_${tmpTableName}_${UUID.randomUUID.toString}_SparkStreamUpload.gz"
+    val blob: CloudBlockBlob = container.getBlockBlobReference(blobName)
+    val gzip: GZIPOutputStream = new GZIPOutputStream(blob.openOutputStream())
     val csvSerializer = new KustoCsvSerializationUtils(schema, timeZone)
 
     val writer = new OutputStreamWriter(gzip, StandardCharsets.UTF_8)
-    val buffer:BufferedWriter = new BufferedWriter(writer, GZIP_BUFFER_SIZE)
-    val csvWriter = new CsvWriter(buffer, new CsvWriterSettings)
+    val buffer: BufferedWriter = new BufferedWriter(writer, GZIP_BUFFER_SIZE)
+    val csvWriter: CsvWriter = new CsvWriter(buffer, new CsvWriterSettings)
+    datasink.FileWriteResource(buffer, gzip, csvWriter, blob)
+  }
 
-    for (row <- rows) {
-      csvWriter.writeRow(csvSerializer.convertRow(row))
+  @throws[IOException]
+  private[kusto] def serializeRows(rows: Iterator[InternalRow],
+                                   storageUri: String,
+                                   parameters: KustoWriteResource): Seq[CloudBlockBlob] = {
+    import parameters._
+    val container: CloudBlobContainer = new CloudBlobContainer(new URI(storageUri))
+    val fileWriter: FileWriteResource = createFileWriter(schema, writeOptions.timeZone, coordinates, tmpTableName, storageUri, container)
+
+    val (_, _, blobs) = rows.foldLeft[(Long, FileWriteResource, Seq[CloudBlockBlob])]((0, fileWriter, Seq())) { case ((size, fileWriter, blobsCreated), row) =>
```

Comment:
Some comments could be helpful here

Reply:
the first one is the size of the last row in the dataframe, and the second one is the last blob that have created, I am not using those parameters after that, so if I give them a name the scalastyle marks them as unused... I can still give them a name and make scalastyle ignore it, what do you think?

Reply:
It's OK not to give names, just add a comment (the first one is the size of the last row in the data frame, and the second one is the last blob that was created)